### PR TITLE
feat: trip details wait section

### PR DESCRIPTION
**Pull request by mortennordseth**
This PR adds information about wait times on the trip details page. 

image

Part of closing https://github.com/AtB-AS/kundevendt/issues/15639

### CODE REVIEW

File: src/page-modules/assistant/details/trip-section/index.tsx [hunk 1]
```diff
@@ -102,30 +121,36 @@ export default function TripSection({
...
+                        interchangeDetails.publicCode,
+                        interchangeDetails.fromPlace,
+                      ),
+                    )
+              }
+            />
+          </TripRow>
+        </div>
+      )}
+      {legWaitDetails?.mustWaitForNextLeg && legWaitDetails.waitTime > 30 && (
```
Comment (mikaelbr) [inline src/page-modules/assistant/details/trip-section/index.tsx:151]: These types of checks can also be named. 
```suggestion
      {showWaitSection && (
```

It makes it easier to read components i think

Comment (mikaelbr) [reply src/page-modules/assistant/details/trip-section/index.tsx:135]: Could also potentially pass in `mustWaitForNextLeg` and have the conditional inside `WaitSection`. Reducing the overhead in reading the component. Drawback is that it isn't as that it is optional, but I think the Component Prop signature with `mustWaitForNextLeg | undefined` can be good enough a indicator

**Review by mikaelbr — CHANGES_REQUESTED** [review]
See comment on simplifying consume of wait time

File: src/page-modules/assistant/details/trip-section/index.tsx (hunk 1 above)
Comment (mortennordseth) [reply src/page-modules/assistant/details/trip-section/index.tsx]: I like this 👍 I actually think that it is easier to read the component using the named check than passing the `mustWaitForNextLeg` to the component 🤔 I'm up for both changes, but I think the way it is now is good

```tsx
{showInterchangeSection && (
  <InterchangeSection
    interchangeDetails={interchangeDetails}
    publicCode={leg.line?.publicCode}
  />
)}
{showWaitSection && (
  <WaitSection waitTimeInSeconds={legWaitDetails.waitTime} />
)}
```

Comment (mikaelbr) [reply src/page-modules/assistant/details/trip-section/index.tsx]: I might have misread as to that `mustWaitForNextLeg` was part of the same `legWaitDetails` object. But yeah, I also think this is fine enough!

Comment (mortennordseth) [reply src/page-modules/assistant/details/trip-section/index.tsx]: It actually is, and when you say it like that I guess it makes more sense 😄 I didn't include the `mustWaitForNextLeg` to the `WaitSection` props as it was not needed, but I guess passing the "entire" object and check if it should return something makes sense

Comment (mortennordseth) [reply src/page-modules/assistant/details/trip-section/index.tsx]: Then again, we should probably do the same with the `InterchangeSection` and pass `guaranteed` as a prop 🤔

File: src/page-modules/assistant/details/index.tsx [hunk 2]
```diff
@@ -70,12 +71,13 @@ export function AssistantDetails({ tripPattern }: AssistantDetailsProps) {
...
             key={index}
             isFirst={index === 0}
-            isLast={index === tripPattern.legs.length}
+            isLast={index === tripPattern.legs.length - 1}
             leg={leg}
             interchangeDetails={getInterchangeDetails(
               tripPattern.legs,
               leg.interchangeTo?.toServiceJourney.id,
             )}
+            legWaitDetails={getLegWaitDetails(leg, tripPattern.legs[index + 1])}
```
Comment (mikaelbr) [inline src/page-modules/assistant/details/index.tsx:80]: This kind of still feels odd for me. Both interchange and this. I don't have any better options for now, but feels kind of fragmented domain logic wise 🤔

**Review by mikaelbr — APPROVED** [review]

File: src/page-modules/assistant/details/trip-section/index.tsx (hunk 1 above)
Comment (mikaelbr) [reply src/page-modules/assistant/details/trip-section/index.tsx]: That might make sense

File: src/page-modules/assistant/details/index.tsx (hunk 2 above)
Comment (mortennordseth) [reply src/page-modules/assistant/details/index.tsx:80]: We might want to handle this server side?

File: src/page-modules/assistant/details/utils.ts [hunk 3]
```diff
@@ -23,3 +24,24 @@ export function formatLineName(
...
+    leg.expectedEndTime,
+    nextLeg.expectedStartTime,
+  );
+  const mustWaitForNextLeg = waitTime > 0;
+
+  return {
+    waitTime,
+    mustWaitForNextLeg,
+  };
+}
```
Comment (mikaelbr) [inline src/page-modules/assistant/details/utils.ts:47]: I think this maybe should be moved to wait section and be made a part of that API as it is so closely coupled. Does it make sense to use it outside that component? if not, it should be colocated

Comment (mortennordseth) [reply src/page-modules/assistant/details/utils.ts:47]: Very true!

File: src/page-modules/assistant/details/index.tsx (hunk 2 above)
Comment (mikaelbr) [reply src/page-modules/assistant/details/index.tsx:80]: Yes! Good point. If this is just data transformation I think it should happen in the service layer.

**Pull request merged**